### PR TITLE
Adds services account auth example to private registry docs

### PR DESCRIPTION
**Pull request by WalkerGriggs**
Doc update to add example of using a GCP service account to authenticate with a GCR.

See https://github.com/kubernetes-sigs/kind/issues/765 and https://github.com/kubernetes-sigs/kind/pull/756#discussion_r310884414

### CODE REVIEW

File: site/content/docs/user/private-registries.md [hunk 1]
```diff
@@ -90,6 +91,24 @@ rm $HOME/.docker/config.json
 [-f $HOME/.docker/config.json.host] && mv $HOME/.docker/config.json.host $HOME/.docker/config.json
 ```
 
+### ... with Service Accounts
```
Comment (neolit123) [inline site/content/docs/user/private-registries.md]: possibly expand the `...` here?

Comment (WalkerGriggs) [reply site/content/docs/user/private-registries.md]: Perhaps `Add Credentials Programmatically with Access Token` and `Add Credentials Programmatically with Service Account`? A little redundant, but we sadly can't nest further than an h3.

Comment (neolit123) [reply site/content/docs/user/private-registries.md]: ok, i see the nesting now.

IMO it should be:

> ## Add Credentials to the Nodes
> ### Mount a Config File to Each Node
> ### Use an  Access Token
> ### Use a Service Account

File: site/content/docs/user/private-registries.md [hunk 2]
```diff
@@ -90,6 +91,24 @@ rm $HOME/.docker/config.json
...
+
+Access tokens are short lived though, so you may prefer to use a Service Account and keyfile.
+First, either download the key from the console or generate one with gcloud:
+
+```
+gcloud iam service-accounts keys create <output.json> --iam-account <account email>
+```
+
+Then, replace the `gcloud auth...` line from the previous snippet with:
```
Comment (neolit123) [inline site/content/docs/user/private-registries.md]: `from the previous section`
?

File: site/content/docs/user/private-registries.md [hunk 3]
```diff
@@ -90,6 +91,24 @@ rm $HOME/.docker/config.json
 [-f $HOME/.docker/config.json.host] && mv $HOME/.docker/config.json.host $HOME/.docker/config.json
 ```
 
+### ... with Service Accounts
+
+Access tokens are short lived though, so you may prefer to use a Service Account and keyfile.
```
Comment (neolit123) [inline site/content/docs/user/private-registries.md]: > Access tokens are short lived, so you may prefer to use a Service Account and keyfile instead.

**Comment by BenTheElder** [comment]
/ok-to-test
thanks for filing this

File: site/content/docs/user/private-registries.md (hunk 1 above)
Comment (BenTheElder) [reply site/content/docs/user/private-registries.md]: > ...  but we sadly can't nest further than an h3.

Not that we should but, why can't we?

Comment (WalkerGriggs) [reply site/content/docs/user/private-registries.md]: @BenTheElder I might actually be wrong. Some markdown interpreters don't like going being h3; I think @neolit123 has the right nesting idea.

Will make changes soon :tm:

File: site/content/docs/user/private-registries.md (hunk 2 above)
Comment (WalkerGriggs) [reply site/content/docs/user/private-registries.md]: Yea, I probably should be more direct and anchor the section

> Then, replace `gcloud auth print-access-token | ...` from the [previous section] (#use-an-access-token) with:

Or maybe you have another way to format the reference without copying the entire snippet?

File: site/content/docs/user/private-registries.md (hunk 1 above)
Comment (BenTheElder) [reply site/content/docs/user/private-registries.md]: FWIW we use [hugo](https://gohugo.io/) and you can view the rendered version at https://deploy-preview-796--k8s-kind.netlify.com/ but @neolit123's suggestion SGTM

**Comment by BenTheElder** [comment]
/lgtm
/approve
thanks a ton for writing this up :-)

**Pull request merged**